tfjs-backend-wasm/src/backend_wasm.ts

```diff
@@ -220,8 +239,18 @@ export async function init(): Promise<{wasm: BackendWasmModule}> {
         factoryConfig.instantiateWasm = createInstantiateWasmFunc(wasmPath);
       }
     }
-    const wasm = simdSupported ? wasmFactorySimd(factoryConfig) :
-                                 wasmFactory(factoryConfig);
+    let wasm: BackendWasmModule;
+    if (threadsSupported && simdSupported) {
+      wasm = wasmFactoryThreadedSimd(factoryConfig);
+      wasm.mainScriptUrlOrBlob = new Blob(
+          [`var _scriptDir = undefined; var WasmBackendModuleSimd = ` +
+           wasmFactoryThreadedSimd.toString()],
+          {type: 'text/javascript'});
+      // } else if (simdSupported) {
```

Comment:
I think you still need this `else if` branch. E.g. Chrome Android supports only SIMD, but not threads